ls1intum/Artemis
# Filter participations without submissions + add submission count to participations

### Checklist

### Motivation and Context
At the moment, it is possible to filter participations to find those with failed submissions (without result). With this PR, participations can also be filtered to find those with no submissions. Additionally, a new column "Submissions" shows how many submissions a participation has (as it is already the case on the exercise scores page).

### Description
- Move transient property `submissionCount` from Result to Participation (`submissionCount` on Result is only used in one occurence and there the participation is also available. Also conceptually, it makes more sense that a Participation knows this information instead of a Result.)
- Add  new query to the StudentParticipationRepository that counts the submissions per participation for an exercise and add a new service method that turns it into a mapping from participation id to submission count
- Use this mapping in the REST controller for `/exercise/{exerciseId}/participations` and set the transient property `submissionCount` accordingly
- In frontend, add the new filter radio button for participations without submissions and add a new column "Submissions" that shows the submission count and links to the overview of these submissions
- For consistency, also link the submission count on the Exercise Scores page to the submission overview page
- Small design improvement: Fix unnecessarily wide columns in participations table that would require horizontal scrolling even for medium-large screen sizes

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Administration
3. Click on Exercises for a course
4. Click on Participations for an exercise
5. Check that new filter option "Show No Submissions" below headline exists and works as one would expect
6. Check that new column "Submissions" exists in table, that the value is correct, that sorting by it works and that it is linked to view these submissions

### Screenshots
![OnPaste 20200209-143419](https://user-images.githubusercontent.com/7109225/74103066-5a6cd880-4b49-11ea-93d4-330c5f7fd38e.png)
![OnPaste 20200209-143455](https://user-images.githubusercontent.com/7109225/74103069-5b9e0580-4b49-11ea-8c83-57505c9a36af.png)
